packages/engine/tests/units/message/remove_self/integration-test.json

```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "steps": 3,
+    "expected-output": {
+      "json-state": {
+        "1": [{}],
```

Comment:
I think we should probably have two agents in this, and have the second one remove itself, just in case.